packages/styles/src/components/accordion.scss

```diff
@@ -1,94 +1,109 @@
+}
 
-  &:last-of-type {
-    border-bottom: 0;
-  }
+.accordion-header {
+  @include type-mixins.font-curve(accordion.$accordion-header-font-curve);
+  font-weight: accordion.$accordion-header-font-weight;
+  line-height: accordion.$accordion-header-line-height;
+  margin: 0;
 }
 
 .accordion-button {
-  @include type.font-curve(accordion.$accordion-font-curve);
+  @include button-mixins.reset-button;
 
-  background-color: transparent;
-  font-weight: accordion.$accordion-button-font-weight;
-  line-height: accordion.$accordion-button-line-height;
+  width: 100%;
+  position: relative;
+  padding-block: accordion.$accordion-button-padding;
+  padding-inline-start: accordion.$accordion-button-padding;
+  padding-inline-end: accordion.$accordion-button-padding + accordion.$accordion-icon-size;
+  text-align: start;
 
-  &:hover,
-  &:focus {
-    &:not(:disabled) {
-      color: accordion.$accordion-button-hover-color;
-    }
+  &:not(:disabled) {
+    cursor: accordion.$accordion-button-cursor;
   }
 
   &:disabled {
-    opacity: button.$btn-disabled-opacity !important;
+    opacity: accordion.$accordion-button-disabled-opacity;
+  }
+
+  &:focus {
+    outline: none;
+    box-shadow: accordion.$accordion-button-focus-box-shadow;
+  }
+
+  &::after {
+    @include icon-mixins.pi(accordion.$accordion-icon-name);
+    content: '';
+    display: block;
+    height: accordion.$accordion-icon-size;
+    width: accordion.$accordion-icon-size;
+    background-size: accordion.$accordion-icon-size;
+    background-position: center;
+    background-repeat: no-repeat;
+    position: absolute;
+    inset-inline-end: accordion.$accordion-button-padding;
+    inset-block-start: 50%;
+    transform: translateY(-50%);
+    transition: accordion.$accordion-icon-transition;
+
+    // TODO: find a way to style icon within web component on dark background
+    @include color-mixins.on-dark-background() {
+      filter: invert(1);
+    }
   }
 
-  .accordion-item[open] &::after {
-    transform: accordion.$accordion-icon-transform;
+  &.collapsed::after {
+    transform: translateY(-50%) accordion.$accordion-icon-transform;
   }
 
-  // style to improve hcm mode
-  @include utilities.high-contrast-mode() {
+  // TODO: find a way to style WHCM within web component
+  @include utility-mixins.high-contrast-mode() {
     &:hover,
     &:focus-visible {
       &:not(:disabled) {
-        outline: spacing.$size-line solid Highlight;
+        outline: accordion.$accordion-border-width solid Highlight;
       }
     }
 
     &:disabled {
       opacity: 1 !important;
     }
+  }
+}
 
-    &::after {
-      background-image: $accordion-hcm-icon;
-    }
-
-    &:not(.collapsed)::after {
-      background-image: $accordion-hcm-icon;
+.accordion-body {
+  padding: accordion.$accordion-body-padding;
+  border-block-start: accordion.$accordion-border-width solid accordion.$accordion-border-color;
+
+  // TODO: find a way to style headings within web component
```

Comment:
Is it tracked somewhere? I mean, those TODO are too often forgotten. Or perhaps you intended to work on it after this PR? Or is there already an established process for those?

Reply:
The `:part()` could be good here and there should be a ticket if it's not in the scope of this PR.

Reply:
I fixed the 3 TODOs that were in this file: 
- "find a way to style icon within web component on dark background": I reintroduced the mask-image for the accordion button so that the icon color is always inherited from the button color
- "find a way to style WHCM within web component": this was not a problem, media queries are taken into account by the shadow DOM
- "find a way to style headings within web component": we decided not to style specifi slotted element from web components

Reply:
FYI For those headings, I didn't know, but it seems that you can override style by using !important: https://www.matuzo.at/blog/2023/pros-and-cons-of-shadow-dom/#con-styling-slotted-content